src/main/java/org/opensearch/ad/model/ValidationAspect.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.ad.model;
+
+import java.util.Collection;
+import java.util.Set;
+
+import org.opensearch.ad.Name;
+import org.opensearch.ad.constant.CommonName;
+
+public enum ValidationAspect implements Name {
+    DETECTOR(CommonName.DETECTOR),
```

Comment:
How about adding some java doc for these validation aspects?

Reply:
added